http1_server.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+import socket
+import email.utils
+
+
+def response_ok(request):
+    response_code = "HTTP/1.1 200 OK"
+    date = "Date: {}".format(email.utils.formatdate(usegmt=True))
+    headers = "Content-Type: Text/HTML"
+    response = "{}\r\n{}\r\n{}\r\n\r\n".format(response_code, date, headers)
+    return response
+
+
+def response_error(code, reason):
+    response_code = "HTTP/1.1 {} {}".format(code, reason)
+    date = "Date: {}".format(email.utils.formatdate(usegmt=True))
+    headers = "Content-Type: Text/HTML"
+    response = u"{}\r\n{}\r\n{}\r\n\r\n".format(response_code, date, headers)
+    return response.encode('utf-8')
+
+
+def parse_request(request):
+    request = request.split('\r\n')
+    first_line = request[0].split(' ')
+    if first_line[0] != "GET":
+        return response_error(405, "Method Not Allowed")
+    elif first_line[2] != "HTTP/1.1":
+        return response_error(505, "HTTP Version Not Supported")
+    else:
+        return response_ok(first_line[1])
```

Comment:
It is my fault due to the way I've worded the spec, but the spec asks for this function to return a URI or raise an error.  This does neither.  Instead you are returning fully built HTTP responses.  This will not work so well when we move to step 2